resnet.py

```diff
@@ -12,16 +12,21 @@
     AveragePooling2D
 )
 from keras.layers.normalization import BatchNormalization
+from keras.regularizers import l2
 from keras import backend as K
 
+# Helper to build a BN -> relu block
+def _bn_relu(input):
+    norm = BatchNormalization(mode=0, axis=CHANNEL_AXIS)(input)
+    return Activation("relu")(norm)
+
 
 # Helper to build a conv -> BN -> relu block
 def _conv_bn_relu(nb_filter, nb_row, nb_col, subsample=(1, 1)):
     def f(input):
         conv = Convolution2D(nb_filter=nb_filter, nb_row=nb_row, nb_col=nb_col, subsample=subsample,
-                             init="he_normal", border_mode="same")(input)
-        norm = BatchNormalization(mode=0, axis=CHANNEL_AXIS)(conv)
-        return Activation("relu")(norm)
+                             init="he_normal", border_mode="same", W_regularizer=l2(0.0001))(input)
```

Comment:
Done. FWIW, I don't really think this makes the builder any more flexible since it's very specifically targeted to build just the architectures from the papers, rather than to allow those architectures as special cases of a more generic builder. Not that that's really a problem, just why I didn't try to be more flexible the first time around.